userlevel/click.cc

```diff
@@ -461,6 +426,58 @@ void do_set_affinity(pthread_t p, int cpu) {
 # define do_set_affinity(p, cpu) /* nothing */
 #endif
 
+// hotswapping
+
+static Router *hotswap_router;
+static Router *hotswap_thunk_router;
+static bool hotswap_hook(Task *, void *);
+static Task hotswap_task(hotswap_hook, 0);
+
+static bool
+hotswap_hook(Task *, void *)
+{
+    hotswap_thunk_router->set_foreground(false);
+    hotswap_router->activate(ErrorHandler::default_handler());
+    router->unuse();
+#if HAVE_MULTITHREAD
+    for (int i = 0; i < other_threads.size(); ++i) {
+        pthread_cancel(other_threads[i]);
```

Comment:
I dont like the pthread_cancel here much, but I found no way to signal the RouterThread drivers to stop before joining the threads. @kohler do you have a better suggestion to stop the drivers before joining?